docs/proposals/metadata.md

```diff
@@ -0,0 +1,152 @@
+   For example, as an user I can say I want to have 'mysql' service available in
+   my application, so the platform will check the ImageRepositories for the
+   Image that have `PUBLISHES="mysql/tcp/3128"` and add it to my deployment.
+
+4. As an author of the Docker image, I want to specify what _secrets_ the image
+   I'm building require, so I can consume them in the Image.
+   For example I can specify `SECRETS=api/twitter` that should tell the platform
+   to lookup for the 'api/twitter' secret and make it available inside the
+   Image.
+
+5. As an author of the Docker image, I need to have a way to disable
+   'clustering' for my Image, in case the Docker image does not support this.
+   In other words, if the Docker image specify this, the ReplicationController
+   generated for this Image should never get `replicas` set higher than 1.
+
+## The list of proposed metadata
+
+### ACCEPT_TAR_STREAM
+
+If this variable is set, then the BuildConfig generated for this Docker image
+will default the build strategy to 'STI'. The Docker image entrypoint is
+expected to support the 'tar' input stream with the application.
+
+### PUBLISHES
+
+This variable contains a list of services that the Docker image publishes. The
+service is described as `<name>/PROTO/PORT` (eg. `mysql/tcp/3128`) where the
+'PROTO' part is optional and default to 'tcp'.
+
+When the platform consumes this Docker image, it should automatically setup
+Service(s) with given name and port, the Route should be created for this
+service and the health-check should be generated for the ReplicationController
+that implements this Service.
+
+Once this Docker image is imported into ImageRepository, the service might be
+referenced by other metadata (SUBSCRIBES, REQUIRES, COMPANIONS, see bellow.)
+
+
+### SUBSCRIBES
+
+This variable contains a list of services that the Docker image can consume.
+The value can be a valid Docker image name or the service name. In order to
+distinguish between them, the proposed format for the value us following:
+
+```
+SUBSCRIBES="service:redis/6379,image:openshift/redis"
+```
+
+In this case, we lookup user ImageRepositories for the Docker image that
+publishes `redis/6379` service. If there is no match, we continue to search for
+the `openshift/redis` image.
+
+The generation tools should automatically generate corresponding Service for
+each defined service.
+
+### REQUIRES
+
+This variable contains a list of services that the Docker image requires in
+order to work properly. The value format is same as for SUBSCRIBES.  Where the
+SUBSCRIBES setup the infrastructure for pub/sub the REQUIRES might indicate that
+the Docker image requires another Docker image in order to work.  For example, a
+'phpmyadmin' requires 'mysql' service to be available, otherwise it won't work
+properly. A special 'backup' Docker image for mysql will require 'mysql' to be
+available within a Pod in order to do database dump from it.
```

Comment:
What about 5.1.  What about percona?  What about a mysql slave?  We've gone down this road before, there's no magic solution.

----- Original Message -----
> > +In this case, we lookup user ImageRepositories for the Docker image that
> > +publishes `redis/6379` service. If there is no match, we continue to
> > search for
> > +the `openshift/redis` image.
> > +
> > +The generation tools should automatically generate corresponding Service
> > for
> > +each defined service.
> > +
> > +### REQUIRES
> > +
> > +This variable contains a list of services that the Docker image requires
> > in
> > +order to work properly. The value format is same as for SUBSCRIBES.  Where
> > the
> > +SUBSCRIBES setup the infrastructure for pub/sub the REQUIRES might
> > indicate that
> > +the Docker image requires another Docker image in order to work.  For
> > example, a
> > +'phpmyadmin' requires 'mysql' service to be available, otherwise it won't
> > work
> > +properly. A special 'backup' Docker image for mysql will require 'mysql'
> > to be
> > +available within a Pod in order to do database dump from it.
> 
> "which of these images is a mysql DB?"  the one that PUBLISHES a mysql
> service.
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/906/files#r24210849